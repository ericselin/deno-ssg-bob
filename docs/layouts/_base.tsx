/** @jsx h */

import { Component, getPath, h, readContents } from "../../mod.ts";

const baseCss = getPath(import.meta.url, "_base.css");

export const Base: Component = async (
  { children },
  { needsCss },
) => (
  <html lang="en">
    <head>
      <meta charSet="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta http-equiv="X-UA-Compatible" content="ie=edge" />
      <title>bob the static site builder</title>
      <link rel="preconnect" href="https://fonts.gstatic.com" />
      <link
        href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&family=Source+Code+Pro:wght@400;700;900&display=swap"
        rel="stylesheet"
      />
      <style>
        {await readContents([baseCss, ...needsCss])}
      </style>
    </head>

    <body>
      <header>
        <a href="/">bob</a>
        <ul>
          <li>
            <a href="/getting-started/">Getting started</a>
          </li>
          <li>
            <a href="/scenarios/">Common scenarios</a>
          </li>
        </ul>
      </header>
      {children}
      <footer>
        Static sites should be fast, including updating content
      </footer>
    </body>
  </html>
);
