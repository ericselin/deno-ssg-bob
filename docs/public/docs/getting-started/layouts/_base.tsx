/** @jsx h */

import { Component, h } from "../../../../../mod.ts";

export const Base: Component = (props) => (
  <html>
    <head>
      <title>This is my new site!</title>
    </head>
    <body>
      <header>
        Main header
      </header>
      {props.children}
    </body>
  </html>
);
