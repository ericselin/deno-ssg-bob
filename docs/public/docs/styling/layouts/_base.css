body {
  font-family: serif;
  margin: 1rem auto;
  max-width: 50rem;
}
