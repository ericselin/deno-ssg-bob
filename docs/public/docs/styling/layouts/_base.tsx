/** @jsx h */

import { Component, h, readContents } from "../../../../../mod.ts";

export const Base: Component = async (
  { children },
  { needsCss },
) => (
  <html lang="en">
    <head>
      <title>This is my page</title>
      <style>
        {await readContents(needsCss)}
      </style>
    </head>
    <body>
      {children}
    </body>
  </html>
);

// Tip! You can add CSS files to any component, also this one
Base.needsCss = "_base.css";
