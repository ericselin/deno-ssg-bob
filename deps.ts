/*
Copyright 2021 Eric Selin

This file is part of `bob`.

`bob` is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

`bob` is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with `bob`. If not, see <https://www.gnu.org/licenses/>.

Please contact the developers via GitHub <https://www.github.com/ericselin>
or email eric.selin@gmail.com <mailto:eric.selin@gmail.com>
*/

export {
  assert,
  assertEquals,
} from "https://deno.land/std@0.109.0/testing/asserts.ts";
export * as path from "https://deno.land/std@0.109.0/path/mod.ts";
export * as yaml from "https://deno.land/std@0.109.0/encoding/yaml.ts";
export { exists } from "https://deno.land/std@0.109.0/fs/exists.ts";
export { walk } from "https://deno.land/std@0.109.0/fs/walk.ts";
export { expandGlob } from "https://deno.land/std@0.109.0/fs/expand_glob.ts";
export type { WalkEntry } from "https://deno.land/std@0.109.0/fs/walk.ts";
export { Marked as md } from "https://deno.land/x/markdown@v2.0.0/mod.ts";
export { parse as parseFlags } from "https://deno.land/std@0.110.0/flags/mod.ts";

import * as log from "https://deno.land/std@0.109.0/log/mod.ts";
export { log };
export type Logger = typeof log;

export { serve } from "https://deno.land/std@0.121.0/http/server.ts";
