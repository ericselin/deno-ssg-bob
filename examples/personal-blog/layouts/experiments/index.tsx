/** @jsx h */

import { Component, h, sortDateDesc } from "../deps.ts";
import { Base } from "../_base.tsx";

const C: Component<any, { title: string }, { title: string }> = (
  _props,
  { page: { content, title }, wantedPages },
) => (
  <Base>
    <h1>{title}</h1>
    <main>
      {wantedPages?.sort(sortDateDesc).map((experiment) => (
        <article>
          {experiment.content}
          <a
            href={experiment.location.url.pathname.replace("README/", "")}
            class="button-link"
          >
            ðŸ”¥ See it in action ðŸ”¥
          </a>
        </article>
      ))}
      {content}
    </main>
  </Base>
);

C.wantsPages = "experiments/*/README.md";

export default C;
