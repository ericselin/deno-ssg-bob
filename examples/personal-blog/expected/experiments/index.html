<html lang="en"><head><meta charSet="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta http-equiv="X-UA-Compatible" content="ie=edge"><title>Experiments | ericselin.dev</title><link rel="preconnect" href="https://fonts.gstatic.com"><link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&family=Source+Code+Pro:wght@400;700;900&display=swap" rel="stylesheet"><style>:root {
  --nord0: #2e3440;
  --nord1: #3b4252;
  --nord2: #434c5e;
  --nord3: #4c566a;
  --nord4: #d8dee9;
  --nord5: #e5e9f0;
  --nord6: #eceff4;
  --nord7: #8fbcbb;
  --nord8: #88c0d0;
  --nord9: #81a1c1;
  --nord10: #5e81ac;
  --nord11: #bf616a;
  --nord12: #d08770;
  --nord13: #ebcb8b;
  --nord14: #a3be8c;
  --nord15: #b48ead;

  --margin: 1rem;
  --max-width: min(600px, 100% - 4 * var(--margin));

  --color-bg: var(--nord0);
  --color-fg: var(--nord6);
  --color-body: var(--nord1);
  --color-indent: var(--nord3);
  --color-muted: var(--nord4);
  --color-accent: var(--nord8);

  --font-text: 'Inter', sans-serif;
  --font-heading: 'Source Code Pro', monospace;
  --font-mono: 'Source Code Pro', monospace;
}

@media (min-width: 500px) {
  :root {
    --margin: 1.5rem;
  }
}

@media (min-width: 768px) {
  :root {
    --margin: 2rem;
  }
}

body {
  padding: 0;
  margin: 0;
  font-family: var(--font-text);
  background: var(--color-body);
  color: var(--color-fg);
}

body > header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: var(--margin);
  color: var(--color-muted);
  font-family: var(--font-heading);
}

body > header > a {
  text-decoration: none;
  font-weight: 900;
  font-size: 2rem;
  background: var(--color-accent);
  width: 2rem;
  height: 2rem;
  line-height: 1.5rem;
  text-align: center;
  border-radius: 25%;
  color: var(--color-body);
}

body > header > a:hover {
  color: var(--color-bg);
}

body > header ul {
  display: grid;
  grid-auto-flow: column;
  gap: 1rem;
  list-style: none;
  padding: 0;
  margin: 0;
}

body > h1 {
  max-width: var(--max-width);
  margin: calc(2 * var(--margin)) auto;
  padding: 0 var(--margin);
  color: var(--color-fg);
  font-family: var(--font-heading);
}

main {
  max-width: var(--max-width);
  margin: var(--margin) auto;
  padding: 0 var(--margin);
  background: var(--color-bg);
  border-radius: 1rem;
  line-height: 1.25;
}

main::before,
main::after {
  content: ' ';
  display: block;
  contain: layout;
  margin: var(--margin) 0;
}

h1 a,
h2 a,
h3 a,
h4 a,
h5 a,
h6 a {
  text-decoration: none;
}

a {
  color: inherit;
  transition: color ease-in 200ms;
}

a:hover {
  color: var(--color-accent);
}

pre {
  max-width: 100%;
  display: block;
  overflow: auto;
}

code {
  font-family: var(--font-mono);
  line-height: initial;
}

article + article {
  margin-top: calc(2 * var(--margin));
}

article + article::before {
  content: '';
  display: block;
  height: 1px;
  width: 50%;
  background: var(--color-indent);
  margin: 0 auto calc(2 * var(--margin));
}

img {
  display: block;
  max-width: 100%;
}

.button-link {
  border: 1px solid var(--color-accent);
  border-radius: 0.5rem;
  padding: 0.5rem 1rem;
  display: inline-block;
  text-decoration: none;
}

body > footer {
  margin: var(--margin);
  text-align: center;
  color: var(--color-muted);
  font-family: var(--font-heading);
}
</style></head><body><header><a href="/">e</a><ul><li><a href="/blog/">Blog</a></li><li><a href="/experiments/">Experiments</a></li></ul></header><h1>Experiments</h1><main><article><h1 id="google-cast-demo">Google Cast demo</h1>
<p>Google Cast is pretty cool. Especially Chromecast, of course. You can cast pretty much any media 
on this super cheap device. But even better, you can build your own apps that run on the Chromecast.</p>
<p>To build an app, you need to create:</p>

<ul>
<li>A &quot;sender&quot; app which controls the casting</li>
<li>A &quot;receiver&quot; app that runs on the Chromecast</li>
</ul>
<p>Also, to test all of this in development, you need to register your device for development (super easy!).</p>
<p><a href="https://developers.google.com/cast/docs/developers">Docs</a></p>
<a href="/experiments/castme/" class="button-link">🔥 See it in action 🔥</a></article><article><h1 id="dragger">dragger</h1>
<p>A study in dragging and dropping in the browser.</p>
<a href="/experiments/dragger/" class="button-link">🔥 See it in action 🔥</a></article><article><h1 id="simple-text-editor">Simple text editor</h1>
<p>This is a test of the Filesystem API. It&#39;s a very simple text editor. Use at your own risk... 😏</p>
<p>The Filesystem API is freaking amazing. With it you can magically open <em>local</em> files and also save them back! 
The API is in &quot;Origin trial&quot; from Chrome 78 onwards, and available as a flag in previous versions.</p>
<p>The API is pretty damn cool, if you ask me. This is a quite nice looking text editor with a contenteditable
<code>pre</code> tag, and the code is not many lines long. Soon the days of native desktop apps are over! 😝</p>
<h2 id="technologies-used">Technologies used</h2>

<ul>
<li>Filesystem API</li>
</ul>
<h2 id="see-more">See more</h2>
<p><a href="https://developers.google.com/web/updates/2019/08/native-file-system">Google developers post</a></p>
<a href="/experiments/editmyfile/" class="button-link">🔥 See it in action 🔥</a></article><article><h1 id="myday">Myday</h1>
<p>See your Google calendar events for today.</p>
<p><a href="https://myday.ericselin.dev">Working domain</a></p>
<h2 id="requirements">Requirements</h2>

<ul>
<li>Google Cloud Platform project with Calendar API enabled</li>
<li>API key and OAuth client for the project</li>
<li>Private top-level domain for hosting (see note below)</li>
</ul>
<p>Note: You need to authorize the OAuth client for a particular URL.
This URL needs to be on a whitelisted domain, which in turn needs
to be a private top-level domain.</p>
<p>This project also showcases using Google Sign-In to request for
additional permissions on demand (incremental authorization). Note
that it&#39;s apparently not possible to bypass the account selection
when doing this.</p>
<p><strong>The Calendar API is restricted and requires validation by Google
before usage. It still works without validation, but shows a very
scary warning.</strong></p>
<p><a href="https://developers.google.com/calendar/quickstart/js">Quick start docs</a></p>
<a href="/experiments/myday/" class="button-link">🔥 See it in action 🔥</a></article><article><h1 id="pwa-weather-app">PWA Weather App</h1>
<p>An installable PWA that shows todays weather compared to yesterday.</p>
<a href="/experiments/weather/" class="button-link">🔥 See it in action 🔥</a></article></main><footer>There's no such thing as the cloud - it's just someone else's computer.</footer></body></html>